chore(python): Various small Python build and dependency fixes

SyntaxColoring opened the pull request:
# Overview

Fix various small things about our Python build system and dev environments that I encountered while working on #15706.

# Test Plan

Make sure CI keeps passing.

# Changelog

See the individual commit messages.

# Risk assessment

Low.

SyntaxColoring committed:
Fix system-server missing from Makefile.

abr-testing, hardware-testing, performance-metrics, and test-data-generation are also missing, but that makes sense because they're special things that we treat as sort of off to the side.

SyntaxColoring committed:
Unify jsonschema constraints.

shared-data/python's setup.py's constraint did not overlap with its Pipfile's constraint, causing `pipenv lock --dev` to fail.

Separately, api's exact version number pin did not match shared-data's exact version number pin. buildroot uses the one that api specifies, so this changes shared-data's to match that.

SyntaxColoring committed:
Mark systemd-python as Linux-only in performance-metrics.

This was preventing `pipenv lock` and `make setup` on macOS.

SyntaxColoring committed:
Update hardware's pydantic constraint.

This did not match shared-data's constraint, causing `pipenv lock` failures.

DerekMaggio reviewed (approved):
LGTM, nice catch on the linux only thing

SyntaxColoring commented:
CI shows linting failing in `hardware/` because of the Pydantic fix (?). Looking into it.

SyntaxColoring committed:
Work around "Missing named argument" errors.

Something about the Pydantic upgrade was interacting poorly with Mypy. Despite these fields having defaults, Mypy was raising errors when the class was instantiated without them. Somehow, declaring the defaults as keyword arguments instead of positional arguments fixes it.

Also fix a typo, `descript` -> `description`.
